src/database/db_post.js

```diff
@@ -0,0 +1,44 @@
+const connect = require('./db_connect');
+
+const hashPassword = request('./../helpers/hashPassword');
+
+const post = {};
+
+post.memory = (newMemory, callback) => {
+  connect.query(
+    `WITH new_memory AS (
+      INSERT INTO memories
+        (user_id, heading, tag, memory)
+      VALUES
+      ((SELECT users.id FROM users WHERE users.username = $1), $2, $3, $4)
+      RETURNING id
+    )
+    UPDATE media SET
+      ${newMemory.mediaType} = ${newMemory.mediaType} || (SELECT id FROM new_memory)`,
```

Comment:
Okay, you should not inject values directly into the query like this. It leaves you vulnerable to **sql injection attacks**